.github/workflows/linux.yml

```diff
@@ -38,3 +38,52 @@ jobs:
         env:
           MAX_WARNINGS: ${{ matrix.conf.max_warnings }}
         run:  ./scripts/count-warnings.py build.log
+
+  build_linux_release_dynamic:
+    name: Release build (dynamic)
+    runs-on: ubuntu-16.04
+    steps:
+      - uses: actions/checkout@v1
+      - run:  sudo apt-get update
+      - name: Install C++ compiler and libraries
+        run:  sudo apt-get install -y tree $(./scripts/list-build-dependencies.sh -m apt -c gcc)
+      - name: Log environment
+        run:  ./scripts/log-env.sh
+      - name: Build
+        run: |
+          set -x
+          export VERSION=$(git describe --abbrev=4)
+          sed -i "s/AC_INIT(dosbox,git)/AC_INIT(dosbox,$VERSION)/" configure.ac
+          ./autogen.sh
+          ./configure CXXFLAGS="-O3 -DNDEBUG"
```

Comment:
Is the `-DNDEBUG` flag the one that `build.sh` is missing?

Reply:
I think it might be missing, yes (but I was not sure what kind of builds you do, so didn't want to modify them right now). I am a bit wary of `-Ofast` because of GCC documentation ("It also enables optimizations that are not valid for all standard-compliant programs."). I omitted `-ffunction-sections` and `-fdata-sections` because:

           Only use these options when there are significant benefits from doing
           so.  When you specify these options, the assembler and linker create
           larger object and executable files and are also slower.  These options
           affect code generation.  They prevent optimizations by the compiler and
           assembler using relative locations inside a translation unit since the
           locations are unknown until link time.  An example of such an
           optimization is relaxing calls to short call instructions.

I think we might end up turning on LTO (which relates to `-ffunction-sections` and `-fdata-sections`, I believe?) - but we probably should estabilish some benchmarks to make informed decisions.

Perhaps we'll end up with several different "Release" targets?

Reply:
Excellent; yes - I can add `-DNDEBUG` to build.sh in my current CI update branch. 

One set of build flags should work; but yes - some benchmarks will figure out if we need to go the extra mile or not. For now, -O3 is going to get us 95% of the way there, and perhaps FDO or LTO to squeeze out a big extra - but no need sweating that right now. 

`-Ofast` is an offshoot from `ffast-math`, which for some math code can _really_ help a lot. It's a slight relaxation on IEEE math strictness and order of operations that result in final rounding errors being larger than IEEE allows, so instead of a float calculation landing .9999999999997  close to 1.0, it becomes .9999999999961.   Similarly, if there is a concern that float rounding is too excessive, you can still switch to double's and -Ofast will still give you way more bits of accuracy than float math without -Ofast.

For our purposes in DOSBox, (almost?) all floating points calculations get rounded back to integers (be it 8 or 16-bit pixels colors, 10-bit pixel locations, or 16-bit audio amplitudes); so having 31.2-bits of accuracy instead of 31.7-bits for our floating points calculations will not impact us.

The new native [bfloat16](https://en.wikipedia.org/wiki/Bfloat16_floating-point_format) format is very exciting because we can double or quadruple our floating point performance, and free up more cycles for the dynrec core.

Yes, when you combine `-ffunction-sections`, `-fdata-sections`, and LDFLAGS=`--as-needed`, the linker will delete all unused function sections and data sections that aren't actually called.  So it can result in a very small binary, especially when you link statically (versus statically linking without those options).